File: mssd-backend/src/main/java/com/mssd/service/ReservationService.java
package com.mssd.service;

import com.mssd.dto.ReservationDto;
import com.mssd.dto.ReservationRequestDto;

import java.time.LocalDateTime;
import java.util.List;

public interface ReservationService {
    List<ReservationDto> getAllReservations();
    ReservationDto getReservationById(Long id);
    List<ReservationDto> getReservationsByCalendarId(Long calendarId);
    List<ReservationDto> getReservationsByVisitorEmail(String visitorEmail);
    List<ReservationDto> getReservationsByStatus(String status);
    ReservationDto createReservation(ReservationRequestDto dto);
    ReservationDto updateReservation(Long id, ReservationRequestDto dto);
    ReservationDto confirmReservation(Long id);
    ReservationDto cancelReservation(Long id);
    void deleteReservation(Long id);
    boolean isReservationPossible(Long calendarId, int numberOfPeople);
} 

File: mssd-frontend/src/app/services/reservation.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

export interface ReservationDto {
  id: number;
  calendarId: number;
  calendarTitle: string;
  visitorName: string;
  visitorEmail: string;
  visitorPhone: string;
  numberOfPeople: number;
  notes?: string;
  status: string;
  reservationDate?: string;
  createdAt: string;
  updatedAt?: string;
}

export interface ReservationRequestDto {
  calendarId: number;
  visitorName: string;
  visitorEmail: string;
  visitorPhone?: string;
  numberOfPeople: number;
  notes?: string;
}

@Injectable({ providedIn: 'root' })
export class ReservationService {
  private apiUrl = '/api/reservations';

  constructor(private http: HttpClient) {}

  createReservation(reservation: ReservationRequestDto): Observable<ReservationDto> {
    return this.http.post<ReservationDto>(this.apiUrl, reservation);
  }

  getAllReservations(): Observable<ReservationDto[]> {
    return this.http.get<ReservationDto[]>(this.apiUrl);
  }

  confirmReservation(id: number): Observable<ReservationDto> {
    return this.http.put<ReservationDto>(`${this.apiUrl}/${id}/confirm`, {});
  }
  cancelReservation(id: number): Observable<ReservationDto> {
    return this.http.put<ReservationDto>(`${this.apiUrl}/${id}/cancel`, {});
  }
  updateReservation(id: number, reservation: ReservationRequestDto): Observable<ReservationDto> {
    return this.http.put<ReservationDto>(`${this.apiUrl}/${id}`, reservation);
  }
  deleteReservation(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }
} 